src/CMakeLists.txt

```diff
@@ -871,16 +871,14 @@ add_subdirectory(compressor)
 
 add_subdirectory(tools)
 
-# dmClock
-
-add_subdirectory(dmclock) # after gmock
-add_dependencies(tests dmclock-tests dmclock-data-struct-tests)
+# dmClock (after gmock)
 
+add_subdirectory(dmclock/src)
 if(WITH_TESTS)
-  install(PROGRAMS
-    ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/dmclock-tests
-    ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/dmclock-data-struct-tests
-    DESTINATION bin)
+  # note: add_test is not being called, so dmclock tests aren't part
+  # of ceph tests
+  add_subdirectory(dmclock/test)
```

Comment:
> I will give this PR a try and let you know the result.

Here [1] is the result of building this PR - if you click on the link at [1] and look at the area on the right where it says "Build Results". . . In that area, under the heading "ceph" are the WITH_TESTS=OFF builds and the builds under "ceph-test" are WITH_TESTS=ON. And you're right - the build does succeed in the newer openSUSE Tumbleweed, but we also need it to succeed in Leap 42.3, which is exhibiting the https://paste2.org/paJcefcb failure.

When you click on any build result (e.g. "failed") it takes you to a page showing the last few lines of the build log. From there you can click on another link to get the full build log, at the beginning of which it shows which packages/versions are installed in the build environment. Possibly the different result is caused by different versions of cmake in Leap 42.3 and Tumbleweed?

It's interesting that the WITH_TESTS=ON build succeeds in CentOS 7.3 - maybe the failure is reproducible in CentOS 7.2, though?

[1] https://build.opensuse.org/package/show/home:smithfarm:branches:filesystems:ceph:luminous/ceph